python/pyarrow/tests/test_types.py

```diff
@@ -53,6 +53,31 @@
 ]
 
 
+@pytest.mark.parametrize('klass', [
+    pa.Field,
+    pa.Schema
+])
+def test_constructor_errors(klass):
+    msg = ("Do not call {cls}'s constructor directly, use `pyarrow.{func}` "
+           "instead.".format(cls=klass.__name__, func=klass.__name__.lower()))
+
+    with pytest.raises(TypeError, match=msg):
+        klass()
+
+
+@pytest.mark.parametrize('klass', [
+    pa.DataType,
+    pa.TimestampType,
+    # others are not part of the public API
```

Comment:
Is this intentional? It looks assymetric.

Reply:
I don't know. If it's not too much work you could test all of them.

Reply:
I mean others are not exposed. Should I expose them?

Reply:
Who do you mean with "expose" exactly?

Reply:
I'm testing under `pa.lib.*Type` instead of `pa.*Type`.

Reply:
Ah, I see. I wonder what the rationale is for exposing `TimestampType`. @xhochy

Reply:
I cannot remember why. @wesm any ideas? I think we only use `pa.timestamp()` at the moment from an end-user perspective.